### PR TITLE
Use TCP port 10110 by default

### PR DESCRIPTION
**Pull request by da2x**
Standard nmea-0183 service port as registered with IANA.

Resolves issue #9.

### CODE REVIEW

**Comment by zeenix** [comment]
Good stuff but as you can see this breaks the test in CI so you'll need to also update the testcase for this.

**Comment by da2x** [comment]
The tests randomly pass/fail because. In my defense, they all passed when I submitted this.

The Linux kernel fails to release the TCP port lock on all interfaces after each test is run and before the next one is started. Previously, each test used different ports so this wasn’t ever an issue.

I’m uncertain about how to proceed here. If I remove `test_stdin_gps_autoport_iface` in favor of `test_stdin_gps_with_port_iface` then all tests pass (as only one will use the default port – requires fixing test `test_stdin_gps_with_port_iface` to actually use a specific port which seems like a bug in the existing test).

**Comment by zeenix** [comment]
hmm.. Can't we ensure that we release the port lock somehow?

**Comment by da2x** [comment]
Oh! Tests are run concurrently. That would explain the issue.

**Comment by da2x** [comment]
Tests can either be run with `cargo test -- --test-threads=1` or I can change the tests to use different ports. I don’t see any other option.

**Comment by zeenix** [comment]
different ports is better. If the service works on one port, it really should work on another as well.

File: src/cmdline_config.rs [hunk 1]
```diff
@@ -39,13 +39,13 @@ pub fn config_from_cmdline() -> Config {
...
                                 .long("--port")
-                                .help("Port to run TCP service on")
+                                .help("Port to run TCP service on (default: 10110)")
                                 .takes_value(true)
                                 .value_name("PORT"))
                            .arg(Arg::with_name("interface")
                                 .short("n")
                                 .long("--network-interface")
-                                .help("Bind specific network interface")
+                                .help("Bind specific network interface (default: all)")
```
Comment (zeenix) [inline src/cmdline_config.rs]: this change is isn't part of the logical change of this patch. I know it's just one line but I'm strict on "one logical change == one patch" rule. :) Feel free to put both these help improvements into the same patch on top of these two.

File: src/cmdline_config.rs [hunk 2]
```diff
@@ -42,7 +42,7 @@ pub fn config_from_cmdline() -> Config {
             Arg::with_name("port")
                 .short("p")
                 .long("--port")
-                .help("Port to run TCP service on")
+                .help("Port to run TCP service on (default: 10110)")
```
Comment (zeenix) [inline src/cmdline_config.rs]: We completely lost the other change. :( I just wanted it in a separate patch.

**Comment by zeenix** [comment]
Do you want me to merge this already and add the lost change myself?

**Comment by da2x** [comment]
I just forgot to push the second commit, sorry. :)

**Comment by zeenix** [comment]
No worries. Congratulations on becoming a valued contributor! :)

**Comment by da2x** [comment]
I’ve got a new Contributor badge and everything. 👍

**Pull request merged**